### PR TITLE
FOUR-15339: Improve saved search sorting performance

### PR DESCRIPTION
**Pull request by runyan-co**
## Issue
Sorting for saved searches with a large results (generally 1000+ tasks) are slow (3+ secs) when sorted.

### Reproduction Steps
1. Log in 
2. Go to Tasks page
3. Have 1000 or more tasks
4. Filters works to status column
5. Create a saved search
6. Use Filters to status column

## Solution
There were duplicate queries for selecting and counting the results of the saved search query, deduplicating them improve performance.

## How to Test
Resort the same saved search and notice the improved speed.

## Related Tickets & Packages
- [FOUR-15339](https://processmaker.atlassian.net/browse/FOUR-15339)
- [package-savedsearch PR](https://github.com/ProcessMaker/package-savedsearch/pull/438)

#### CICD
ci:next
ci:deploy
ci:package-savedsearch:bugfix/FOUR-15339

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

.

[FOUR-15339]: https://processmaker.atlassian.net/browse/FOUR-15339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-39e4651b94.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-39e4651b94.engk8s.processmaker.net

**Review by marcoAntonioNina — APPROVED** [review]

File: ProcessMaker/Http/Controllers/Api/TaskController.php [hunk 1]
```diff
@@ -269,8 +278,11 @@ public function update(Request $request, ProcessRequestToken $task)
     private function handleQueryException($e)
     {
         $regex = '~Column not found: 1054 Unknown column \'(.*?)\' in \'where clause\'~';
+
         preg_match($regex, $e->getMessage(), $m);
 
+        dump($m);
```
Comment (ryancooley) [inline ProcessMaker/Http/Controllers/Api/TaskController.php:284]: @runyan-co Is this meant to remain here?

**Review by ryancooley — CHANGES_REQUESTED** [review]

File: ProcessMaker/Http/Controllers/Api/TaskController.php (hunk 1 above)
Comment (runyan-co) [reply ProcessMaker/Http/Controllers/Api/TaskController.php]: Whoops, haha, I just pushed a commit to remove it. 👍🏻

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-39e4651b94.engk8s.processmaker.net

**Pull request merged**